### ISSUE TITLE
Task::Star bailing out with failing DBIish

### ISSUE THREAD

**Issue by dha** [perl6/DBIish#65]
Trying to install Task::Star, and when it gets to DBIish, I get a huge wad of error messages. Pasting below. Rakudo and Panda built from the github repositories (as of commits a31ab3b84bd7852f42ab71fcf16ed787aa470efb

and

cb273cd55be0554796c801c72dfa3833e5270938

respectively. OS X 10.10.5)

Errors:

# Failed test 'Can install driver for 'SQLite''
# at t/01-Basic.t line 35
 
Use of uninitialized value $drv of type Any in string context
Any of .^name, .perl, .gist, or .say can stringify undefined things, if needed.  in block  at t/01-Basic.t line 38
 
# Failed test 'Is an instance '''
# at t/01-Basic.t line 38
 
# Failed test 'Any indeed a driver'
# at t/01-Basic.t line 39
 
# Failed test 'version declared'
# at t/01-Basic.t line 42
Method 'version' not found for invocant of class 'Any'
  in block <unit> at t/01-Basic.t line 43
 
# Looks like you planned 47 tests, but ran 30
# Looks like you failed 4 tests of 30
t/01-Basic.t ..............
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 21/47 subtests
t/03-lib-util.t ........... ok
t/05-mock.t ............... ok
 
  in block  at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 54
  in any  at /Users/dha/panda/.panda-work/1463002570_1/lib/.precomp/AC5B7F0AE411E6FBB0069E88D5318B962FDAD64C.1463000968.72001/7D/7D00A6C693B7DF515CAA5ECC6097F930493C50AA line 1
  in method install-driver at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 58
  in block <unit> at t/10-mysql.t line 67
 
t/10-mysql.t ..............
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 87/87 subtests
 
  in block  at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 54
  in any  at /Users/dha/panda/.panda-work/1463002570_1/lib/.precomp/AC5B7F0AE411E6FBB0069E88D5318B962FDAD64C.1463000968.72001/7D/7D00A6C693B7DF515CAA5ECC6097F930493C50AA line 1
  in method install-driver at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 58
  in method run-tests at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish/CommonTesting.pm6 (DBIish::CommonTesting) line 33
  in block <unit> at t/25-mysql-common.t line 5
 
t/25-mysql-common.t .......
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 107/107 subtests
 
  in block  at t/26-mysql-blob.t line 16
  in any  at /Users/dha/panda/.panda-work/1463002570_1/lib/.precomp/AC5B7F0AE411E6FBB0069E88D5318B962FDAD64C.1463000968.72001/7D/7D00A6C693B7DF515CAA5ECC6097F930493C50AA line 1
  in block  at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 41
  in any  at /Users/dha/panda/.panda-work/1463002570_1/lib/.precomp/AC5B7F0AE411E6FBB0069E88D5318B962FDAD64C.1463000968.72001/7D/7D00A6C693B7DF515CAA5ECC6097F930493C50AA line 1
  in any  at /Users/dha/panda/.panda-work/1463002570_1/lib/.precomp/AC5B7F0AE411E6FBB0069E88D5318B962FDAD64C.1463000968.72001/7D/7D00A6C693B7DF515CAA5ECC6097F930493C50AA line 1
  in block  at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 54
  in any  at /Users/dha/panda/.panda-work/1463002570_1/lib/.precomp/AC5B7F0AE411E6FBB0069E88D5318B962FDAD64C.1463000968.72001/7D/7D00A6C693B7DF515CAA5ECC6097F930493C50AA line 1
  in method install-driver at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 58
  in method connect at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 44
  in block <unit> at t/26-mysql-blob.t line 11
 
t/26-mysql-blob.t .........
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 19/19 subtests
 
  in block  at t/27-mysql-datetime.t line 16
  in any  at /Users/dha/panda/.panda-work/1463002570_1/lib/.precomp/AC5B7F0AE411E6FBB0069E88D5318B962FDAD64C.1463000968.72001/7D/7D00A6C693B7DF515CAA5ECC6097F930493C50AA line 1
  in block  at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 41
  in any  at /Users/dha/panda/.panda-work/1463002570_1/lib/.precomp/AC5B7F0AE411E6FBB0069E88D5318B962FDAD64C.1463000968.72001/7D/7D00A6C693B7DF515CAA5ECC6097F930493C50AA line 1
  in any  at /Users/dha/panda/.panda-work/1463002570_1/lib/.precomp/AC5B7F0AE411E6FBB0069E88D5318B962FDAD64C.1463000968.72001/7D/7D00A6C693B7DF515CAA5ECC6097F930493C50AA line 1
  in block  at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 54
  in any  at /Users/dha/panda/.panda-work/1463002570_1/lib/.precomp/AC5B7F0AE411E6FBB0069E88D5318B962FDAD64C.1463000968.72001/7D/7D00A6C693B7DF515CAA5ECC6097F930493C50AA line 1
  in method install-driver at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 58
  in method connect at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 44
  in block <unit> at t/27-mysql-datetime.t line 11
 
t/27-mysql-datetime.t .....
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 12/12 subtests
t/30-pg.t ................. ok
# DBDish::Pg: Can't connect: could not connect to server: No such file or directory
#   Is the server running locally and accepting
#   connections on Unix domain socket "/var/pgsql_socket/.s.PGSQL.5432"?
#  (1)
# Can't continue.
t/35-pg-common.t .......... ok
t/36-pg-array.t ........... ok
t/36-pg-blob.t ............ ok
t/36-pg-native.t .......... ok
t/37-pg-datetime.t ........ ok
 
  in block  at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 54
  in any  at /Users/dha/panda/.panda-work/1463002570_1/lib/.precomp/AC5B7F0AE411E6FBB0069E88D5318B962FDAD64C.1463000968.72001/7D/7D00A6C693B7DF515CAA5ECC6097F930493C50AA line 1
  in method install-driver at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 58
  in method run-tests at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish/CommonTesting.pm6 (DBIish::CommonTesting) line 33
  in block <unit> at t/40-sqlite-common.t line 6
 
t/40-sqlite-common.t ......
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 107/107 subtests
 
  in block  at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 54
  in any  at /Users/dha/panda/.panda-work/1463002570_1/lib/.precomp/AC5B7F0AE411E6FBB0069E88D5318B962FDAD64C.1463000968.72001/7D/7D00A6C693B7DF515CAA5ECC6097F930493C50AA line 1
  in method install-driver at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 58
  in block <unit> at t/41-sqlite-exec-error.t line 13
 
# Looks like you planned 16 tests, but ran 2
t/41-sqlite-exec-error.t ..
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 14/16 subtests
 
  in block  at t/42-sqlite-blob.t line 18
  in any  at /Users/dha/panda/.panda-work/1463002570_1/lib/.precomp/AC5B7F0AE411E6FBB0069E88D5318B962FDAD64C.1463000968.72001/7D/7D00A6C693B7DF515CAA5ECC6097F930493C50AA line 1
  in block  at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 41
  in any  at /Users/dha/panda/.panda-work/1463002570_1/lib/.precomp/AC5B7F0AE411E6FBB0069E88D5318B962FDAD64C.1463000968.72001/7D/7D00A6C693B7DF515CAA5ECC6097F930493C50AA line 1
  in any  at /Users/dha/panda/.panda-work/1463002570_1/lib/.precomp/AC5B7F0AE411E6FBB0069E88D5318B962FDAD64C.1463000968.72001/7D/7D00A6C693B7DF515CAA5ECC6097F930493C50AA line 1
  in block  at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 54
  in any  at /Users/dha/panda/.panda-work/1463002570_1/lib/.precomp/AC5B7F0AE411E6FBB0069E88D5318B962FDAD64C.1463000968.72001/7D/7D00A6C693B7DF515CAA5ECC6097F930493C50AA line 1
  in method install-driver at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 58
  in method connect at /Users/dha/panda/.panda-work/1463002570_1/lib/DBIish.pm6 (DBIish) line 44
  in block <unit> at t/42-sqlite-blob.t line 13
 
t/42-sqlite-blob.t ........
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 18/18 subtests
# DBIish: DBDish::Oracle needs 'libclntsh.12.1.dylib', not found
# Can't continue.
t/45-oracle-common.t ...... ok
t/46-oracle-blob.t ........ ok
t/47-oracle-datetime.t .... ok
 
Test Summary Report
-------------------
t/01-Basic.t            (Wstat: 65280 Tests: 30 Failed: 4)
  Failed tests:  27-30
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 47 tests but ran 30.
t/03-lib-util.t         (Wstat: 0 Tests: 6 Failed: 0)
  TODO passed:   5-6
t/10-mysql.t            (Wstat: 65280 Tests: 0 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 87 tests but ran 0.
t/25-mysql-common.t     (Wstat: 65280 Tests: 0 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 107 tests but ran 0.
t/26-mysql-blob.t       (Wstat: 65280 Tests: 0 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 19 tests but ran 0.
t/27-mysql-datetime.t   (Wstat: 65280 Tests: 0 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 12 tests but ran 0.
t/40-sqlite-common.t    (Wstat: 65280 Tests: 0 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 107 tests but ran 0.
t/41-sqlite-exec-error.t (Wstat: 65280 Tests: 2 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 16 tests but ran 2.
t/42-sqlite-blob.t      (Wstat: 65280 Tests: 0 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 18 tests but ran 0.
Files=19, Tests=362, 66 wallclock secs ( 0.12 usr  0.04 sys + 57.50 cusr  4.61 csys = 62.27 CPU)
Result: FAIL
The spawned process exited unsuccessfully (exit code: 1)
  in sub run-and-gather-output at /Users/dha/rakudo/install/share/perl6/site/sources/24811C576EF8F85E7672B26955C802BB2FC94675 (Panda::Common) line 85
  in block  at /Users/dha/rakudo/install/share/perl6/site/sources/48E2EB9144E069353B240AD2D147B48C65F70152 (Panda::Tester) line 22
  in sub indir at /Users/dha/rakudo/install/share/perl6/site/sources/24811C576EF8F85E7672B26955C802BB2FC94675 (Panda::Common) line 20
  in method test at /Users/dha/rakudo/install/share/perl6/site/sources/48E2EB9144E069353B240AD2D147B48C65F70152 (Panda::Tester) line 5
  in method install at /Users/dha/rakudo/install/share/perl6/site/sources/582CB7486602954A4601BDCE5A0EAC54B05DA58A (Panda) line 156
  in block  at /Users/dha/rakudo/install/share/perl6/site/sources/582CB7486602954A4601BDCE5A0EAC54B05DA58A (Panda) line 229
  in method resolve at /Users/dha/rakudo/install/share/perl6/site/sources/582CB7486602954A4601BDCE5A0EAC54B05DA58A (Panda) line 223
  in sub MAIN at /Users/dha/rakudo/install/share/perl6/site/resources/E0D978079BB5081DE986D058BB8AB08252F05CC8 line 18
  in block <unit> at /Users/dha/rakudo/install/share/perl6/site/resources/E0D978079BB5081DE986D058BB8AB08252F05CC8 line 152
 
~/panda 17:37:17%

**Comment by salortiz** [comment]
@dha  Found that Task::Star somehow fails to include `NativeHelpers::Blob` required by `DBIish`.
Just to confirm that hypothesis, can you try to install with `panda` or `zef` that, and then retry the install of `Task::Start` or `DBIish`?

Thank you for the report.

**Comment by dha** [comment]
On Wed, May 11, 2016 at 09:56:08PM -0700, Salvador Ortiz wrote:
> @dha  Found that Task::Star somehow fails to include `NativeHelpers::Blob` required by `DBIish`.
> Just to confirm that hypothesis, can you try to install with `panda` or `zef` that, and then retry the install of `Task::Start` or `DBIish`?

Sorry I didn't respond sooner, been running around getting ready for
OSCON. In fact, I'm in the "introducing Perl 6" tutorial right now. :-)

FWIW, I tried installing DBIish again, and this time it passed. Then I
tried installing NativeHelpers::Blob, and it appears to be installed,
although I did not get around to explicitly installing it.

So maybe this got fixed while I was indisposed.

> Thank you for the report.

Thank you for making it work, however that happened. :-)

dha

-- 
David H. Adler - <dha@pobox.com> - http://www.panix.com/~dha/
"scanf() is evil."
